database-commons/src/main/java/io/cdap/plugin/db/CommonSchemaReader.java

```diff
@@ -129,11 +129,10 @@ public boolean isTypeCompatible(Schema.Field field, ResultSetMetaData metadata,
     int precision = metadata.getPrecision(index);
     int scale = metadata.getScale(index);
 
-    Schema.Type fieldType = field.getSchema().isNullable() ? field.getSchema().getNonNullable().getType()
-      : field.getSchema().getType();
-    Schema columnSchema = DBUtils.getSchema(type, precision, scale);
-    Schema.Type columnType = columnSchema.getType();
+    Schema inputFieldSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
+    Schema outputFieldSchema = DBUtils.getSchema(type, precision, scale);
 
-    return Objects.equals(fieldType, columnType);
+    return Objects.equals(inputFieldSchema.getType(), outputFieldSchema.getType()) &&
```

Comment:
why do you think nullable value from inputschema equals to non-nullable value from output schema?